tests/server/resource_deploy_performance.py

```diff
@@ -0,0 +1,316 @@
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+
+import asyncio
+import datetime
+import json
+import uuid
+
+from pyformance import global_registry
+
+from inmanta import const
+from inmanta.data.model import AttributeStateChange
+from inmanta.util import get_compiler_version
+
+
+async def test_resource_deploy_performance(server, client, clienthelper, environment, agent, caplog):
+    """
+    Method to 'quickly' benchmark the deploy performance wrt to the database
+
+    Not run by default, because filename doesn't start with test
+    """
+    n_version = 5
+    n_resources = 1000
+    n_deploys = 5
+
+    env_id = uuid.UUID(environment)
+
+    start = datetime.datetime.now()
+
+    for model_version in range(1, n_version + 1):
+        version = await clienthelper.get_version()
+
+        resources = [
+            {
+                "key": f"key{resource}",
+                "value": "value1",
+                "id": f"test::Resource[agent1,key=key{resource}],v={version}",
+                "send_event": False,
+                "purged": False,
+                "requires": [],
+            }
+            for resource in range(1, n_resources + 1)
+        ]
+
+        result = await client.put_version(
+            tid=env_id,
+            version=version,
+            resources=resources,
+            resource_state={},
+            unknowns=[],
+            version_info={},
+            compiler_version=get_compiler_version(),
+        )
+        assert result.code == 200
+
+        for deploy in range(1, n_deploys + 1):
+            # run in parallel a bit
+
+            batches = 10
+            per_batch = int(n_resources / batches)
+
+            async def run_batch(index):
+                for resource in range((per_batch * index) + 1, (per_batch * (index + 1) + 1)):
+                    action_id = uuid.uuid4()
+                    rvid = f"test::Resource[agent1,key=key{resource}],v={version}"
+                    result = await agent._client.resource_deploy_start(tid=env_id, rvid=rvid, action_id=action_id)
+                    assert result.code == 200, result.result
+
+                    result = await agent._client.resource_did_dependency_change(tid=environment, rvid=rvid)
+                    assert result.code == 200, result.result
+
+                    result = await agent._client.resource_deploy_done(
+                        tid=env_id,
+                        rvid=rvid,
+                        action_id=action_id,
+                        status=const.ResourceState.deployed,
+                        messages=[],
+                        changes={"attr1": AttributeStateChange(current=None, desired="test")},
+                        change=const.Change.created,
+                    )
+                    assert result.code == 200, result.result
+
+            await asyncio.gather(*[run_batch(i) for i in range(0, batches)])
+            end = datetime.datetime.now()
```

Comment:
I think for benchmarks [time.process_time](https://docs.python.org/3/library/time.html#time.process_time) is the recommended measurement. I'm not sure about how large a difference it makes, might be negligible.

Reply:
In this particular case, I'm more interested in wallclock time: I also want to take into account the time taken by postgresql. 

I could go for [time.monotonic()](https://docs.python.org/3/library/time.html#time.monotonic), if you want.